tests/conftest.py

```diff
@@ -289,85 +289,78 @@ def display_recipient_factory(recipient_details_list: List[Tuple[int, str]]):
     ]
 
 
-stream_msg_template = {
-    "id": 537286,
-    "sender_full_name": "Foo Foo",
-    "timestamp": 1520918722,
-    "client": "website",
-    "sender_email": "foo@zulip.com",
-    "type": "stream",
-    "sender_realm_str": "",
-    "flags": ["read"],
-    "sender_id": 5140,
-    "content_type": "text/x-markdown",
-    "stream_id": 205,
-    "subject": "Test",
-    "reactions": [],
-    "subject_links": [],
-    "avatar_url": "dummy_avatar_url",
-    "is_me_message": False,
-    "content": "Stream content here.",
-    "display_recipient": "PTEST",
-}
+def msg_template_factory(
+    msg_id: int,
+    msg_type: str,
+    timestamp: int,
+    *,
+    stream_id: Optional[int] = None,
+    recipients: Union[str, List[Dict[str, Any]]] = "PTEST",
+):
+    """
+    Generate message template for all types of messages(stream/PM/group)
+    """
+    return {
+        "id": msg_id,
+        "sender_full_name": "Foo Foo",
+        "timestamp": timestamp,
+        "client": "website",
+        "sender_email": "foo@zulip.com",
+        "type": msg_type,
+        "sender_realm_str": "",
+        "flags": ["read"],
+        "sender_id": 5140,
+        "content_type": "text/x-markdown",
+        "stream_id": stream_id,
+        "subject": "",
+        "reactions": [],
+        "subject_links": [],
+        "avatar_url": "dummy_avatar_url",
+        "is_me_message": False,
+        "content": "Stream content here.",
```

Comment:
I have customized the content internally, depending on `msg_type`. I don't see the need for having an extra parameter for this field, especially since we don't test on this and probably we shouldn't?

Reply:
I was initially concerned that all the versions would imply a stream message based on the content in your previous version, which you've resolved :+1:

We don't need precise control to support the refactor you've done here, but if we reuse these fixtures later rather than setting the content directly we might provide support in the factory "initializer".